### PR TITLE
Portal UI server logs for missing session secret fix small TFM mapping bug

### PR DESCRIPTION
**Pull request by ttbarnes**
- add logs for when session secret env var is missing
- remove `SESSION_SECRET` crypto fallback as it doesn't work
- Also fixed TFM facility currency mapping so it only maps currencyCode if currency object exists.

### CODE REVIEW

File: gef-ui/server/index.js [hunk 1]
```diff
@@ -20,49 +19,54 @@ const app = express();
...
-const SESSION_SECRET = process.env.SESSION_SECRET || crypto.randomBytes(256 / 8).toString('hex');
+if (!process.env.SESSION_SECRET) {
+  console.error('GEF UI server - SESSION_SECRET missing');
+}
+
+if (!process.env.REDIS_HOSTNAME) {
+  console.error('GEF UI server - REDIS_HOSTNAME missing');
+}
+
+if (!process.env.REDIS_KEY) {
```
Comment (pricea1) [inline gef-ui/server/index.js:30]: I think REDIS_KEY is optional an only needed for Azure envs. Works fine locally if not configured.

File: gef-ui/server/index.js [hunk 2]
```diff
@@ -20,49 +19,54 @@ const app = express();
 
 const PORT = process.env.PORT || 5006;
 
-// Fail-safe fallback to a 256-bit random value:
-const SESSION_SECRET = process.env.SESSION_SECRET || crypto.randomBytes(256 / 8).toString('hex');
+if (!process.env.SESSION_SECRET) {
+  console.error('GEF UI server - SESSION_SECRET missing');
+}
+
+if (!process.env.REDIS_HOSTNAME) {
```
Comment (pricea1) [inline gef-ui/server/index.js:26]: We also get a connection error on line 53 is REDIS_HOSTNAME is invalid. Do we need this too?

File: portal/server/index.js [hunk 3]
```diff
@@ -21,47 +20,52 @@ const app = express();
 
 const PORT = process.env.PORT || 5000;
 
-// Fail-safe fallback to a 256-bit random value:
-const SESSION_SECRET = process.env.SESSION_SECRET || crypto.randomBytes(256 / 8).toString('hex');
+if (!process.env.SESSION_SECRET) {
```
Comment (pricea1) [inline portal/server/index.js:23]: Same comments as gef-ui

**Review by pricea1 — COMMENTED** [review]
A couple of queries.

**Comment by ttbarnes** [comment]
Good points @pricea1 i've removed the console logs for missing Redis env vars. So now the server changes is just logging out if the session secret is missing, and simplifies some conditions.

**Review by pricea1 — APPROVED** [review]
LGTM

**Pull request merged**